mlir/lib/Analysis/DataFlow/IntegerRangeAnalysis.cpp

```diff
@@ -43,6 +43,9 @@ LogicalResult staticallyNonNegative(DataFlowSolver &solver, Value v) {
   if (!result || result->getValue().isUninitialized())
     return failure();
   const ConstantIntRanges &range = result->getValue().getValue();
+  if (!range.umin().getBitWidth() || !range.umax().getBitWidth() ||
```

Comment:
I feel like a decent contract for this function would be to assert that `isa<IntegerType>(v.getType())`, but I don't have a strong opinion here

Reply:
I think the general shake of the solution is to seal all the isUninitialized()s for a new isUnknown() or isNonInteger()

Reply:
... wow that's a lot of autocorrect-os

Hopefully the overall message got through

The other thing that'd be needed there is to make sure that whenever we've visiting a value, we always make the result something other than "unknown"